### DISCUSSION TITLE
Download Mode File - Exclude Everything, Then Whitelist

### DISCUSSION THREAD

**Discussion by damnski** [gomods/athens#1788] [Q&A] [answered]
Hello,

I have use case question for the Download Mode File (https://docs.gomods.io/configuration/download/).  We'd like to exclude everything, then whitelist only certain modules.  Is this possible?  If it is possible, another side effect is how to add this to the use cases in the documentation?

Thank you,
Best,
-dkw

**Comment by ngshiheng** [comment, marked as answer]
hey @damnski ! it is possible, here's a simple example of a Download Mode File:

```hcl
downloadURL = "https://proxy.golang.org"

mode = "sync"

download "company.com/allow/a" {
    mode = "sync"
}

download "company.com/allow/b" {
    mode = "sync"
}

download "company.com/allow/*" {
    mode = "none"
}
```

In the download mode file above, only `company.com/allow/a` and `company.com/allow/b` will be whitelisted.

However, do note that even downloads are restricted, the `/list` and `/latest` handlers will still return data to the proxy user. I'm not sure if this is intended but just a note here.

Reference: https://github.com/gomods/athens/pull/1826, https://github.com/gomods/athens/issues/1774
